ext/openssl/ossl_ssl.c

```diff
@@ -2429,6 +2429,35 @@ ossl_ssl_alpn_protocol(VALUE self)
 	return rb_str_new((const char *) out, outlen);
 }
 
+/*
+ * call-seq:
+ *    session.export_keying_material(label, length) -> String
+ *
+ * Enables use of shared session key material in accordance with RFC 5705.
+ */
+static VALUE
+ossl_ssl_export_keying_material(VALUE self, VALUE label, VALUE length)
+{
+    SSL *ssl;
+    VALUE str;
+    unsigned char *p;
+    size_t len = NUM2INT(length);
+    int use_ctx = 0;
+    unsigned char *ctx;
+    size_t ctx_len = 0;
+
+    GetSSL(self, ssl);
+
+    str = rb_str_new(0, len);
+    p = (unsigned char *)RSTRING_PTR(str);
+    int ret = SSL_export_keying_material(ssl, p, len, (unsigned char *)RSTRING_PTR(label),
+					 RSTRING_LENINT(label), ctx, ctx_len, use_ctx);
```

Comment:
Please move the declaration of `ret` to the beginning of the function. Unfortunately, we still have to write in C89 style as long as we support Ruby 2.6.